### PR TITLE
dev/fitchbe/doc 9385

### PR DESCRIPTION
**Pull request by fitchbe**
Move some "About" and "Get Started" content out of the readme into the documentation, add a link in the readme to the documentation, and perform copy editing on the affected content.

### CODE REVIEW

**Comment by mmeterel** [comment]
@fitchbe We are about to make considerable changes to README with #144. Could we rebase this PR after merging #144?

**Comment by fitchbe** [comment]
@mmeterel Sure. Worst case, I can replace the PR with a new one if the rebasing proves too unwieldy.

File: docs/build_setup.rst
Comment (andrewtbarker) [inline docs/build_setup.rst]: Should we link to `selecting_a_compiler.rst` here?

**Review by andrewtbarker — APPROVED** [review]
Overall this looks fine to me. I agree with Mesut that we should maybe wait until the rocBLAS README is merged to merge this.

File: docs/build_setup.rst
Comment (fitchbe) [reply docs/build_setup.rst]: Done; thanks.

**Comment by fitchbe** [comment]
Closing and submitting a new one that's up-to-date with the rocBLAS changes.

**Pull request closed without merging**
